### PR TITLE
NAS-132307 / 25.04 / remove accepts from CacheService plugin

### PR DESCRIPTION
**Pull request by yocalebo**
Our `api_method` decorator was designed (primarily) for public facing endpoints. However, for a private service like the `CacheService`, there is no reason to go through the boilerplate of adding all the arguments/returns. It's much easier and better (imo) to simply keep it marked private and remove the `@accepts` decorator. I've used builtin annotations to keep some semblance of what argument types are being passed to each method. This was easiest path forward without breaking backwards compatibility. There should be no change in functionality.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/cache.py [hunk 1]
```diff
@@ -1,81 +1,58 @@
...
 
+class CacheService(Service):
     class Config:
         private = True
 
     def __init__(self, *args, **kwargs):
         super(CacheService, self).__init__(*args, **kwargs)
         self.__cache = {}
-        self.kv_tuple = namedtuple('Cache', ['value', 'timeout'])
+        self.kv_tuple = namedtuple("Cache", ["value", "timeout"])
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/cache.py:16]: Why are we changing quotes here?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cache.py:16]: Just standardized the file to double-quotes

File: src/middlewared/middlewared/plugins/cache.py [hunk 2]
```diff
@@ -1,81 +1,58 @@
...
-        if cache is not None:
-            cache = cache.value
-
-        return cache
+    def pop(self, key: str):
+        """Removes and returns `key` from cache."""
+        try:
+            self.__cache.pop(key, None).value
+        except AttributeError:
+            pass
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/cache.py:44]: Where's the return?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cache.py:44]: Fixed it

File: src/middlewared/middlewared/plugins/cache.py [hunk 3]
```diff
@@ -1,81 +1,58 @@
...
-
-        if cache is not None:
-            cache = cache.value
-
-        return cache
+    def pop(self, key: str):
+        """Removes and returns `key` from cache."""
+        try:
+            return self.__cache.pop(key, None).value
+        except AttributeError:
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/cache.py:43]: Would it be better to avoid AttributeError in first place?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cache.py:43]: It's quicker to catch known error then check for existence 🤷 but I can go back to original way of doing it.

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cache.py]: fixed it

**Review by creatorcary — APPROVED** [review]

**Pull request merged**